src/rest_framework_api_key/models.py

```diff
@@ -40,17 +40,19 @@ def create_key(self, **kwargs: typing.Any) -> typing.Tuple["AbstractAPIKey", str
     def get_usable_keys(self) -> models.QuerySet:
         return self.filter(revoked=False)
 
-    def is_valid(self, key: str) -> bool:
+    def get_from_key(self, key: str) -> "AbstractAPIKey":
         prefix, _, _ = key.partition(".")
-
         queryset = self.get_usable_keys()
+        api_key = queryset.get(prefix=prefix)
```

Comment:
Shoul we explicitly re-raise the `DoesNotExist` exception here? Eg…

```suggestion
        try:
            api_key = queryset.get(prefix=prefix)
        except self.model.DoesNotExist:
            raise  # For the sake of being explicit.
```

Reply:
It seems to me that the outcome will be identical, but I like the idea of explicitly re-raising the exception to make the intention of the code clearer!